openedx/core/djangoapps/oauth_dispatch/tests/mixins.py

```diff
@@ -33,25 +33,37 @@ def _decode_jwt(verify_expiration):
             Helper method to decode a JWT with the ability to
             verify the expiration of said token
             """
-            keys = KEYS()
+            key_set = []
```

Comment:
It looks like this code largely duplicates what's in edx-drf-extensions. Can we call that code instead?

(Having the duplicated code increases maintenance burden but also risks us having a mismatch in behavior between the tests and how we actually verify and decode JWTs.)

Reply:
I'm planning to move a JWT-related similar code to this repo https://github.com/edx/token-utils. 
Once I start working on that repo. Will handle that task at that time.

Reply:
It's possible that move makes sense, but it deserves as ADR, and will have some consequences:
- Ownership of token-utils is currently owned by cosmonauts, and I imagine we would need a discussion about that. Would it move to arbi-bom or arch-bom? We'd need to discuss with both teams.
- Yes, if this became a required dependency, it would need to move to the `openedx` repo.
- Would these utilities be exposed via edx-drf-extensions as well, or just through the new repo? Would it depend on whether or not it was specifically JWT auth related?
- The repo itself is not in a great state. It looks like https://github.com/edx/token-utils/blob/main/docs/decisions/0001-purpose-of-this-repo.rst was never updated, and I imagine other docs, so its purpose is not yet clearly documented.

Things to consider.

Reply:
It's hard to say whether a move to token-utils is a good idea without knowing more about what that repo is supposed to be. (It's not clear from the README or other docs.) I am interested in knowing more about it, but it feels out of scope for the present work.

Here's what I was picturing for deduplication: https://github.com/openedx/edx-platform/pull/32466 -- this rewrites the mixin to use Django setting overrides so that it can just call our standard decoders.

Reply:
> It's hard to say whether a move to token-utils is a good idea without knowing more about what that repo is supposed to be. (It's not clear from the README or other docs.)

The repo token-utils was created for edx-exams, where they wanted to create custom signed JWTs (i.e. non-auth JWTs). I'll leave it to you all decide how to proceed, but it is important to note that edx-exams has its own token signing which will need to go through this same process. See https://github.com/edx/edx-internal/blob/2785fc7599e1f39fbd5387c3da18c0e74883f4b2/argocd/applications/edx-exams/prod-config.yaml#L102-L104 for the config. Note: it seems this setting was designed to just allow a single signing key per service for **all** custom signed JWTs. Not sure if that is the best design, but that is certainly not an issue for today.

Reply:
@mumarkhan999: Is there a top-level Github Issue or Confluence page tracking this work across all services where we can document these types of things?

Reply:
@robrap Yes we have one. Here's the link
https://github.com/openedx/edx-drf-extensions/issues/290

The ticket does not have many details for now. But planning to add some there.

Reply:
Thanks. I added a link to that ticket in the PR description. I think it would be useful to add references to the top-level ticket in PR descriptions and/or commit comments so we could find all the parts from the top-level issue as needed.